extensions/spotify-beta/src/search.tsx

```diff
@@ -0,0 +1,165 @@
+import { useState, useEffect } from "react";
+import { Action, ActionPanel, Grid, Icon, List, LocalStorage } from "@raycast/api";
+import { usePromise } from "@raycast/utils";
+import { useSearch } from "./hooks/useSearch";
+import { View } from "./components/View";
+import { ArtistsSection } from "./components/ArtistsSection";
+import { AlbumsSection } from "./components/AlbumsSection";
+import { TracksSection } from "./components/TracksSection";
+import { PlaylistsSection } from "./components/PlaylistsSection";
+import { debounce } from "./helpers/debounce";
+import { ShowsSection } from "./components/ShowsSection";
+import { EpisodesSection } from "./components/EpisodessSection";
+
+const filters = {
+  all: "All",
+  artists: "Artists",
+  tracks: "Songs",
+  albums: "Albums",
+  playlists: "Playlists",
+  shows: "Podcasts",
+  episodes: "Episodes",
+};
+
+type FilterValue = keyof typeof filters;
+
+function SearchCommand() {
+  const {
+    data: recentSearchesData,
+    isLoading: recentSearchIsLoading,
+    revalidate: recentSearchRevalidate,
+  } = usePromise(() => LocalStorage.getItem<string>("recent-searches"));
```

Comment:
Why not using useCachedPromise here?

Reply:
Good point, dont remember why I changed this!